sysutils/smart/src/www/widgets/widgets/smart_status.widget.php

```diff
@@ -40,16 +40,13 @@
     </tr>
 
 <?php
-$devs = array();
-## Get all adX, daX, and adaX (IDE, SCSI, and AHCI) devices currently installed
-exec("ls /dev | grep '^\(ad\|da\|ada\)[0-9]\{1,2\}$'", $devs); ## From SMART status page
+$devs = preg_split ("/[\s]+/", trim(configd_run ("smart list")));
 
 if (count($devs) > 0) {
     foreach ($devs as $dev) {
 ## for each found drive do
-        $dev_ident = exec("diskinfo -v /dev/$dev | grep ident   | awk '{print $1}'"); ## get identifier from drive
-        $dev_state = trim(exec("smartctl -H /dev/$dev | awk -F: '/^SMART overall-health self-assessment test result/ {print $2;exit}
-/^SMART Health Status/ {print $2;exit}'")); ## get SMART state from drive
+        $dev_ident = trim(configd_run ("smart ident /dev/$dev")); ## get identifier from drive
```

Comment:
passes unescaped arguments to configd theres a configdp_run() I think...

furthermore, you want to coalesce both calls while at it if possible? maybe even all three configd calls?